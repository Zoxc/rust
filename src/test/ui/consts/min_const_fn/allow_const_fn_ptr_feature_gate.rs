#![feature(staged_api)]

#[stable(feature = "rust1", since = "1.0.0")]
const fn error(_: fn()) {}

#[stable(feature = "rust1", since = "1.0.0")]
#[rustc_allow_const_fn_ptr]
//~^ ERROR internal implementation detail
const fn compiles(_: fn()) {}
//~^ ERROR rustc_promotable and rustc_allow_const_fn_ptr attributes must be paired

fn main() {}
