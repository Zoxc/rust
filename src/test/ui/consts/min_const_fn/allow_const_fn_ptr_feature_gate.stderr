error[E0658]: internal implementation detail
  --> $DIR/allow_const_fn_ptr_feature_gate.rs:7:1
   |
LL | #[rustc_allow_const_fn_ptr]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/29642
   = help: add `#![feature(rustc_attrs)]` to the crate attributes to enable

error[E0717]: rustc_promotable and rustc_allow_const_fn_ptr attributes must be paired with either a rustc_const_unstable or a rustc_const_stable attribute
  --> $DIR/allow_const_fn_ptr_feature_gate.rs:9:1
   |
LL | const fn compiles(_: fn()) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
